plone/app/z3cform/converters.py

```diff
@@ -228,7 +228,7 @@ def toWidgetValue(self, value):
             return self.field.missing_value
         separator = getattr(self.widget, 'separator', ';')
         if IRelationList.providedBy(self.field):
-            return separator.join([IUUID(o) for o in value if value])
+            return separator.join([IUUID(o) for o in value if value and o])
```

Comment:
Yes, you are right, but this case happened many times to me (Plone 5.0.4):
If you for example already used that widget for a field and related any objects to it and then you delete that objects in Plone value stored for this field still contains this already deleted object. If you try to change this value inside form field - it raises TypeError: ('Could not adapt', None, <InterfaceClass plone.uuid.interfaces.IUUID>) error.

So I'm suggesting to add double the check on every object in "value" itself.